docs/src/wallet_to_wallet_with_tariscript.md

```diff
@@ -0,0 +1,205 @@
+4. Script \\( \alpha_a \\) whose hash \\( \sigma_a = H(\alpha_a) \\),
+5. Script input \\( \theta_a \\) for script \\( \alpha_a \\),
+6. Script key \\( k_{sa} \\) whose public value (\\( K_{sa} = k_{sa} \cdot G \\)) is the key left over when script \\( \alpha_a \\) is executed with input \\( \theta_a \\)   
+7. Height \\( h_a \\) where \\( C_a \\) was mined.
+
+Alice then chooses the following values:
+1. Amount \\( v_b \\) being sent to Bob
+2. Fee value \\( f \\). The amounts need to balance out so \\( v_a = v_b + f \\)    
+3. A random nonce \\( r_a \\) whose public value is \\( R_a = r_a \cdot G \\)
+4. An UTXO offset key \\( k_{Ob} \\) for Bob's UTXO, whose public value is \\( K_{Ob} = k_{Ob} \cdot G \\)
+5. The script \\( \alpha_b \\) that will become the spending script on \\( C_b \\).
+6. Change spending key \\( k_c \\) for the change output \\( C_c \\)
+8. A script for the change UTXO \\( \alpha_c \\)
+9. A script key \\( k_{sc} \\) for \\( \alpha_c \\)    
+10. The UTXO offset key for the change UTXO \\( k_{Oc} \\) whose public value is \\( K_{Oc} = k_{Oc}G \\)
+ 
+Firstly Alice prepares the transaction the same way she would for a vanilla MW transaction by calculating the excess:
+\\( x_s = k_c - k_a \\)
+
+Alice will also calculate the rangeproof \\( RP_c \\) for her change output for the modified commitment
+
+\\( \hat{C}_c = (k_c + \beta_c) \cdot G + v_c \cdot H \\)
+
+where
+
+\\( \beta_c = H(\sigma_c || F_c || K_{Oc}) \\).
+
+Alice then sends the following values to Bob:
+
+| Symbol  | Description                                      |
+|---------|--------------------------------------------------                                   |
+| \\( R_a \\)       | Alice's Public Nonce                                                      |
+| \\( v_b \\)       | Amount being sent to Bob                                                  |
+| \\( f \\)         | fee                                                                       |
+| \\( m \\)         | Transaction metadata (currently just lockheight)                          |
+| \\( \sigma_b \\)  | Hash of \\( C_b \\)'s spending script,  \\( \sigma_b = H(\alpha_b) \\)    |
+| \\( K_{Ob} \\)    | Public UTXO script offset                                                 |
+| message           | A unicode string                                                          |
+
+### Bob Replies
+Once Bob receives these values he will choose the following:
+1. Spending key \\( k_b \\) of \\( C_b \\)
+2. A random nonce \\( r_b \\) whose public value is \\( R_b = r_b \cdot G \\)
+3. His script key \\( k_s \\) whose public value \\( K_s = k_s \cdot G \\) must be the final value on the stack after script \\( alpha_b \\) executes.
+4. Output features \\( F_b \\) for \\( C_b \\) if any.
+
+Bob produces the rangeproof \\(RP_b) for the modified commitment
+
+\\( \hat{C}_b = (k_b + \beta_b) \cdot G + v_b \cdot H \\) 
+
+where 
+
+\\( \beta_b = H(\sigma_b || F_b || K_{Ob}) \\).
```

Comment:
What is K_0b in this case?

Reply:
The Public UTXO Offset key given to Bob by Alice